Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -63,7 +64,8 @@ class OpenSearchEngine: NSObject, NSCoding {
     /**
      * Returns the search URL for the given query.
      */
-    func searchURLForQuery(query: String) -> NSURL? {
+    func searchURLForQuery(query: String, searchTemplate: String? = nil) -> NSURL? {
```

Comment:
I don't understand the changes in this file. If we're doing a keyword "search", we should get the URL template from the bookmarks, then replace the `%s` from the bookmark URL with the search term. We shouldn't need to make any changes to search engines.